src/compiler/compiler.ts

```diff
@@ -1,64 +1,81 @@
-import { FieldInfo } from "../ClassFile/types/fields";
-import { MethodInfo } from "../ClassFile/types/methods";
-import { ConstantPoolManager } from "./constant-pool-manager";
+import * as fs from 'fs'
+import * as peggy from 'peggy'
+import { ClassFile } from '../ClassFile/types'
+import { AST } from '../ast/types/packages-and-modules'
+import {
+  ClassBodyDeclaration,
+  ClassDeclaration,
+  ConstructorDeclaration,
+  FieldDeclaration,
+  MethodDeclaration
+} from '../ast/types/classes'
+import { AttributeInfo } from '../ClassFile/types/attributes'
+import { FieldInfo } from '../ClassFile/types/fields'
+import { MethodInfo } from '../ClassFile/types/methods'
+import { ConstantPoolManager } from './constant-pool-manager'
 import {
   generateClassAccessFlags,
   generateFieldAccessFlags,
-  generateMethodAccessFlags,
-} from "./compiler-utils";
-import { SymbolTable } from "./symbol-table";
-import { generateCode } from "./code-generator";
+  generateMethodAccessFlags
+} from './compiler-utils'
+import { SymbolTable } from './symbol-table'
+import { generateCode } from './code-generator'
 
-const MAGIC = 0xcafebabe;
-const MINOR_VERSION = 0;
-const MAJOR_VERSION = 52;
+const MAGIC = 0xcafebabe
+const MINOR_VERSION = 0
+const MAJOR_VERSION = 52
 
 export class Compiler {
-  private symbolTable: SymbolTable;
-  private constantPoolManager: ConstantPoolManager;
-  private interfaces: Array<number>;
-  private fields: Array<FieldInfo>;
-  private methods: Array<MethodInfo>;
-  private attributes: Array<AttributeInfo>;
-  private className: string;
+  private symbolTable: SymbolTable
+  private constantPoolManager: ConstantPoolManager
+  private interfaces: Array<number>
+  private fields: Array<FieldInfo>
+  private methods: Array<MethodInfo>
+  private attributes: Array<AttributeInfo>
+  private className: string
 
   constructor() {
-    this.setup();
+    this.setup()
   }
 
   private setup() {
-    this.constantPoolManager = new ConstantPoolManager();
-    this.interfaces = [];
-    this.fields = [];
-    this.methods = [];
-    this.attributes = [];
-    this.symbolTable = new SymbolTable();
+    this.constantPoolManager = new ConstantPoolManager()
+    this.interfaces = []
+    this.fields = []
+    this.methods = []
+    this.attributes = []
+    this.symbolTable = new SymbolTable()
   }
 
   compile(ast: AST) {
-    this.setup();
-    this.symbolTable.handleImports(ast.importDeclarations);
-    const classFiles: Array<ClassFile> = [];
-    ast.topLevelClassOrInterfaceDeclarations.forEach(x => classFiles.push(this.compileClass(x)));
-    return classFiles[0];
+    this.setup()
+    this.symbolTable.handleImports(ast.importDeclarations)
+    const classFiles: Array<ClassFile> = []
+    ast.topLevelClassOrInterfaceDeclarations.forEach(x => classFiles.push(this.compileClass(x)))
+    return classFiles[0]
+  }
+
+  compileFromSource(javaProgram: string) {
+    const javaPegGrammar = fs.readFileSync(__dirname + '/__tests__/main.pegjs', 'utf8')
```

Comment:
is this test utility? might be better to move into `__tests__`?

Reply:
No, this function is meant to provide an alternative option to parse the source program using `pegjs` instead of the one provided in `src/ast/parser.ts`. But yeah, the path is incorrect, will fix.